R/multitables.R

```diff
@@ -26,6 +26,43 @@ setMethod("[[", c("MultitableCatalog", "numeric"), function (x, i, ...) {
     getEntity(x, i, Multitable, ...)
 })
 
+#' @rdname catalog-extract
+#' @export
+setMethod("[[<-", c("MultitableCatalog", "character", "missing", "formula"),
+          function (x, i, j, value) {
+              stopifnot(length(i) == 1)
+              
+              ds <- loadDataset(datasetReference(x))
+              template <- formulaToQuery(value, data = ds)
+              
+              if (i %in% names(x)) {
+                  payload <- makeMultitablePayload(template = template)
+                  crPATCH(urls(x)[match(i, names(x))], body=toJSON(payload))
+                  return(x)
+              } else {
+                  ## Creating a new filter
+                  f <- .newMultitable(template, self(x), name = i)
+                  return(refresh(x))
+              }
+          })
+
+makeMultitablePayload <- function (template, ...) {
+    template <- lapply(template$dimensions, function (x) list(query=x))
+
+    return(wrapEntity(template = template, ...))
+}
+
+#' @rdname catalog-extract
+#' @export
+setMethod("[[<-", c("MultitableCatalog", "ANY", "missing", "NULL"),
+          function (x, i, j, value) {
+              stopifnot(length(i) == 1)
+              if (i %in% names(x) | is.numeric(i)) {
```

Comment:
What if `i` isn't in `names(x)` or isn't numeric? Add tests for that.

Consider deferring the `i` validation to L61, where `x[[i]]` will handle it. Then you just have to determine how to handle invalid `i` in this method. `NULL` assignment into R lists seems to generally not error if it doesn't match an existing element in the list:

```
> d <- list()
> d[["a"]] <- NULL
> d[[25]] <- NULL
> d
list()
```